mathesar_ui/src/systems/multi-tagger/MultiTagger.svelte

```diff
@@ -0,0 +1,184 @@
+    );
+    searchBox?.focus?.();
+  }
+
+  async function toggle(index: number) {
+    const option = options.at(index);
+    if (!option) return;
+    option.update((o) => o.asLoading());
+    const { key, mappingIds } = get(option);
+    if (Array.isArray(key)) {
+      throw new Error('Key cannot be an array');
+    }
+    try {
+      if (mappingIds.length === 0) {
+        const newMappingId = await addMapping(controller, key);
+        option.update((o) => o.withMappings([newMappingId]));
+      } else {
+        await removeMapping(controller, mappingIds);
+        option.update((o) => o.withoutMappings());
+      }
+    } catch (error) {
+      toast.error(getErrorMessage(error));
+    }
+    option.update((o) => o.asNotLoading());
+    focusSearch();
+  }
+
+  function selectIndex(i: number) {
+    selectedIndex = i;
+  }
+
+  function selectNext() {
+    if (selectedIndex === undefined) {
+      selectedIndex = 0;
+      return;
+    }
+    selectedIndex = Math.min(selectedIndex + 1, options.length - 1);
+  }
+
+  function selectPrevious() {
+    if (selectedIndex === undefined) {
+      selectedIndex = 0;
+      return;
+    }
+    selectedIndex = Math.max(selectedIndex - 1, 0);
+  }
+
+  function toggleSelected() {
+    if (selectedIndex === undefined) return;
+    void toggle(selectedIndex);
+  }
+
+  function handleKeyDown(e: KeyboardEvent) {
+    const action = new Map([
+      ['Escape', close],
+      ['ArrowUp', selectPrevious],
+      ['ArrowDown', selectNext],
+      ['Enter', toggleSelected],
+    ]).get(e.key);
+    if (!action) return;
+    e.preventDefault();
+    action();
+  }
+</script>
+
+<div id={elementId} tabindex="0" data-multi-tagger>
+  <div class="search">
+    <MultiTaggerSearch {controller} onKeyDown={handleKeyDown} />
+  </div>
+
+  {#if $records.isInitializing}
+    <div class="padded"><Spinner /></div>
+  {:else if $records.error}
+    <ErrorBox>{RpcError.fromAnything($records.error).message}</ErrorBox>
+  {:else}
+    <div class="results">
+      {#each options as option, index}
+        <MultiTaggerRow
+          {option}
+          selected={index === selectedIndex}
+          searchValue={$searchValue}
+          onToggle={() => {
+            void toggle(index);
+          }}
+          onHover={() => selectIndex(index)}
+        />
+      {:else}
+        <div class="no-results padded">{$_('no_records_found')}</div>
+      {/each}
+    </div>
+
+    {#if hasPagination}
+      <div class="footer">
+        <div class="pagination">
+          <MiniPagination
+            bind:pagination={$pagination}
+            on:change={() => controller.getRecords()}
+            recordCount={recordsCount}
+          />
+        </div>
+      </div>
+    {/if}
+  {/if}
+</div>
+
+<style lang="scss">
+  div[data-multi-tagger] {
+    max-width: min(30rem, 90vw);
+    max-height: inherit;
+    overflow: hidden;
+    position: relative;
+    display: grid;
+    grid-template-rows: auto 1fr auto;
+  }
```

Comment:
I find the dropdown height extending till the end of the view port frustrating since the dropdown always ends up displaying on the right/left of the cell instead of below it.

<img width="877" height="921" alt="Image" src="https://github.com/user-attachments/assets/eb6a0e87-815d-4958-8aa6-ef6db9006d37" />

I prefer it appearing below the cell, similar to the row seeker.

<img width="716" height="544" alt="Image" src="https://github.com/user-attachments/assets/badffc8f-2ba2-49a8-91b8-d4d6e78bf9fb" />

While this is subjective and I'm willing to update my views based on your reasoning, I do think that both these components need to present a similar UX, whichever we choose.